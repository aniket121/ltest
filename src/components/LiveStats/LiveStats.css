.LiveIcon {
  border: 1px solid rgba(255, 38, 73, 1);
  background-image: linear-gradient(
    -44deg,
    rgba(255, 2, 64, 1) 0%,
    rgba(255, 2, 64, 1) 100%
  );
}


.MaterialIcon {
  font-size: 16px;
  margin-right: 5px;
}

.PreviewIcon {
  background-image: linear-gradient(
    -44deg,
    rgba(0, 0, 0, 0.3) 0%,
    rgba(0, 0, 0, 0.5) 100%
  );
}

.ViewerIcon {
  background-color: #615f5d;
}

.LiveViewerIcons {
  display: flex;
}

.BroadcastIcon {
  display: flex;
  justify-content: center;
  align-items: center;
  /*font-size: 12px;*/
  padding: 0 12px;
  border-radius: 14px;
  box-shadow: 0px 1px 27px 0px rgba(0, 0, 0, 0.5);
  margin-right: 12px;
}

.BroadcastIcons {
  display: flex;
  position: absolute;
  top: 28px;
  left: 32px;
  z-index: 2;
}

.platformIcons {
  display: flex;
}