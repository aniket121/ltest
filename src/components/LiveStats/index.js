import React from 'react';
import SocialIconLive from 'components/SocialIconLive';
import styles from './LiveStats.css';

const Platforms = ({ currentBroadcast = null }) => {
	if (!currentBroadcast) {
		return null;
	}

	const platforms = currentBroadcast.streamTargetContainers.map(container => container.streamTarget.provider);

	const uniquePlatforms = platforms.filter(function(item, pos) {
		return platforms.indexOf(item) === pos;
	});

	// Uncomment this line and comment above logic
	// for testing live broadcast w/o going live
	// const uniquePlatforms = ['facebook', 'youtube', 'twitch'];

	return (
		<div className={styles.platformIcons}>
			{uniquePlatforms.map((platform, i) => <SocialIconLive provider={platform} key={i} />)}
		</div>
	);
};

const LiveStats = ({ isLive, viewers, ...props }) => {
	let viewerCount = 0,
		isFacebook = false;
	if (viewers) {
		viewerCount = Object.keys(viewers).reduce((sum, source) => sum + viewers[source], 0);
	}

	if (props.currentBroadcast) {
		isFacebook = props.currentBroadcast.streamTargetContainers.some(
			container => container.streamTarget.provider === 'facebook'
		);
	}

	return (
		<div className={styles.BroadcastIcons}>
			{isLive ? (
				<div className={styles.LiveViewerIcons}>
					<div className={styles.LiveIcon + ' ' + styles.BroadcastIcon}>
						<i className={'material-icons ' + styles.MaterialIcon}>fiber_manual_record</i>
						Live
					</div>
					
					{isFacebook && (
						<div className={styles.ViewerIcon + ' ' + styles.BroadcastIcon}>
							<i className={'material-icons ' + styles.MaterialIcon}>remove_red_eye</i>
							<span>{viewerCount}</span>
						</div>
					)}
				</div>
			) : (
				<div className={styles.PreviewIcon + ' ' + styles.BroadcastIcon}>Preview</div>
			)}
			<Platforms {...props} />
		</div>
	);
};

export default LiveStats;
