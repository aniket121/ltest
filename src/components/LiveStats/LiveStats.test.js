import React from 'react';
import { shallow } from 'enzyme';
import LiveStats from 'components/LiveStats';

const createProps = props => ({
	isLive: false,
	currentBroadcast: null,
	viewers: {
		facebook_timeline: 100,
		facebook_page: 400,
	},
	...props,
});

describe('<LiveStats />', () => {
	test('renders snapshot', () => {
		const props = createProps();
		const wrapper = shallow(<LiveStats {...props} />);
		expect(wrapper).toMatchSnapshot();
	});

	test('should render preview icon if not live', () => {
		const props = createProps();
		const wrapper = shallow(<LiveStats {...props} />);
		expect(
			wrapper
				.children()
				.first()
				.text()
		).toBe('Preview');
	});

	test('should render live icon if live', () => {
		const props = createProps({ isLive: true });
		const wrapper = shallow(<LiveStats {...props} />);
		expect(
			wrapper
				.children()
				.first()
				.children()
				.first()
				.children()
				.last()
				.text()
		).toBe('Live');
	});

	test('should render all provider icons by default', () => {
		const props = createProps({ currentBroadcast: {} });
		const wrapper = shallow(<LiveStats {...props} />);
		expect(wrapper.find('Platforms')).toHaveLength(1);
	});

	test('should render number of viewers', () => {
		const props = createProps({
			isLive: true,
			currentBroadcast: {
				streamTargetContainers: [{ streamTarget: { provider: 'facebook' } }],
			},
		});
		const wrapper = shallow(<LiveStats {...props} />);
		expect(
			wrapper
				.children()
				.first()
				.children()
				.last()
				.children()
				.last()
				.text()
		).toBe('500');
	});

	test('should not render number of viewers', () => {
		const props = createProps({
			isLive: true,
			currentBroadcast: {
				streamTargetContainers: [{ streamTarget: { provider: 'youtube' } }],
			},
			viewers: null,
		});
		const wrapper = shallow(<LiveStats {...props} />);
		expect(
			wrapper
				.children()
				.first()
				.children()
				.last()
				.children()
		).toBe('500');
	});
});
