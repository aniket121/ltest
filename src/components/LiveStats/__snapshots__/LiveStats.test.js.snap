exports[`<LiveStats /> renders snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className={undefined}>
    <div
        className="undefined undefined">
        Preview
    </div>
    <Platforms
        currentBroadcast={null} />
</div>,
  "nodes": Array [
    <div
      className={undefined}>
      <div
            className="undefined undefined">
            Preview
      </div>
      <Platforms
            currentBroadcast={null} />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <LiveStats
        currentBroadcast={null}
        isLive={false}
        viewers={
                Object {
                        "facebook_page": 400,
                        "facebook_timeline": 100,
                      }
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "currentBroadcast": null,
          "isLive": false,
          "viewers": Object {
            "facebook_page": 400,
            "facebook_timeline": 100,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className={undefined}>
          <div
                    className="undefined undefined">
                    Preview
          </div>
          <Platforms
                    currentBroadcast={null} />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className={undefined}>
          <div
                    className="undefined undefined">
                    Preview
          </div>
          <Platforms
                    currentBroadcast={null} />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LiveStats
    currentBroadcast={null}
    isLive={false}
    viewers={
        Object {
            "facebook_page": 400,
            "facebook_timeline": 100,
          }
    } />,
}
`;
